OpenSim/Common/ComponentList.h

```diff
@@ -76,6 +78,21 @@ class ComponentFilterMatchAll : public ComponentFilter {
         return new ComponentFilterMatchAll(*this);
     }
 };
+
+/** A component is considered a match if its full path name matches the 
+provided regular expression. */
+class ComponentFilterFullPathNameRegex : public ComponentFilter {
+public:
+    ComponentFilterFullPathNameRegex(const std::regex& expr) : _expr(expr) {}
```

Comment:
If @tkuchida agrees, go ahead and remove regex. I don't know if it worth having or not. If you do want to have regex, using a simple string will be better.